crates/ide/src/navigation_target.rs

```diff
@@ -504,9 +515,14 @@ impl TryToNav for hir::LifetimeParam {
 
 impl TryToNav for hir::ConstParam {
     fn try_to_nav(&self, db: &RootDatabase) -> Option<NavigationTarget> {
-        let InFile { file_id, value } = self.source(db)?;
+        let InFile { file_id, value } = self.merge().source(db)?;
         let name = self.name(db).to_smol_str();
 
+        let value = match value {
+            Either::Left(Either::Right(x)) => x,
+            _ => unreachable!(),
```

Comment:
```suggestion
            _ if never!() => return None,
```

Reply:
I changed it to this:
```rust
_ => {
    never!();
    return None;
}
```
Because `if never!()` won't compile.